src/core/models.py

```diff
@@ -0,0 +1,34 @@
+from django.db import models
+
+
+class Recipe(models.Model):
+    """Recipe model object"""
+    name = models.CharField(max_length=255)
+    description = models.CharField(
+        max_length=255,
+        blank=True)
+
+    def __str__(self):
+        return self.name
+
+
+class Ingredient(models.Model):
+    """Ingredient model object"""
+    name = models.CharField(max_length=255)
+    recipe = models.ForeignKey(
+        Recipe,
+        related_name='ingredients',
+        on_delete=models.CASCADE)
+
+    def save(self, *args, **kwargs):
+        """Overrides save prohibiting saving the same ingredient per recipe"""
+        if self.id is None:
+            results = Ingredient.objects.filter(
+                name=self.name, recipe=self.recipe)
+            if len(results) > 0:
```

Comment:
There are several things here:
1) You could do this query less costy:
```python
if results.exists():
```
this will issue only an `EXISTS` sql query instead of getting the full queryset.

Another option would be to use `results.first()` which will return the first result if there is at least one or return `None`. This could then be used later to assign the id. However,..

2) You have a race condition here if two ingredients are save at the same time. You could add [unique_together](https://docs.djangoproject.com/en/dev/ref/models/options/#django.db.models.Options.unique_together) to the model to make sure at database level that there are no duplicated ingredients per recipe.

Reply:
True, some available options would be creating a lock or based on your recommendation move the responsibility to the DB and use the `unique_together`. 

I moved forward with the `unique_together` and create a migration. Django doesn’t take the `IntegrityError` lightly. I caught the error it in the `add-ingredient` action and disable the creation and update mixings in the ingredient view as there are no immediate use for them. 

Definitely a braking change.